### PR TITLE
Add itercov function

### PR DESCRIPTION
**Pull request by slwatkins**
Added a function for iteratively cutting a multivariate distribution, similar to iterstat. Also made some minor cosmetic changes to iterstat.

In summary:
- Added `qetpy.cut.itercov`
- Changed `print` statements in `qetpy.cut.iterstat` to warnings
- White space fixes
- Added `qetpy.cut.itercov` test
- Updated `qetpy.cut.iterstat` test

### CODE REVIEW

File: qetpy/cut/_cut.py [hunk 1]
```diff
@@ -65,46 +73,155 @@ def iterstat(data,cut=3,precision=1000.0):
...
+            mean_this = np.mean(data, axis=1)
+            cov_this = np.atleast_1d(np.cov(data))
+            mask = np.ones(nevts, dtype=bool)
+
+            return mean_this, cov_this, mask
+
+        mean_this = np.mean(data[:, mask], axis=1)
+        cov_this = np.atleast_1d(np.cov(data[:, mask]))
+
+        if np.any(np.abs(mean_this - mean_last) < err_mean) or np.any(np.abs(cov_this - cov_last) < err_cov):
```
Comment (cwfink) [inline qetpy/cut/_cut.py]: I think there is a typo in : `np.abs(mean_this - mean_last) < err_mean`, I think these should be `>`

File: qetpy/cut/_cut.py [hunk 2]
```diff
@@ -65,46 +73,155 @@ def iterstat(data,cut=3,precision=1000.0):
...
+
+def itercov(*args, nsigma=2.75, threshold=None, maxiter=15, frac_err=1e-3):
+    """
+    Function for iteratively determining the estimated covariance matrix of a multidimensional
+    normal distribution.
+
+    Parameters
+    ----------
+    args : array_like
+        The data to be iteratively cut on. Can be inputted as a single N-by-M array or a
```
Comment (cwfink) [inline qetpy/cut/_cut.py]: typo

File: qetpy/cut/_cut.py (hunk 1 above)
Comment (slwatkins) [reply qetpy/cut/_cut.py]: Fixed in latest commit.

File: qetpy/cut/_cut.py (hunk 2 above)
Comment (slwatkins) [reply qetpy/cut/_cut.py]: Fixed in latest commit.

**Pull request merged**